impress-org/givewp
# Feature/paypal webhooks 4894

Resolves #4894 

## Description

This PR introduces the framework for registering/unregistering and processing events with PayPal Commerce. It does not provide any handling for a specific event (outside of an example), focusing on the framework itself. 

### Flow Notes
- When a new account is connected, the webhook will be automatically created for the account, listening to the appropriate events.
- When an account is disconnected, the existing webhook will be removed from the database and unregistered with PayPal, to prevent events coming in for disconnected accounts.

### Extendibility
The Webhooks repository and the PayPalWebhooks controller can be extended by other plugins to register additional PayPal events and event handlers.

## Affects

This account connection and disconnection process have been hooked into. They should still work as before, but also register/unregister the webhook.

## Pre-review Checklist




## Testing Instructions

Please make sure to try connecting a new account, checking to see if the webhook id shows up in the options table under the `give_paypal_commerce_webhook_id` key.

**NOTE:** The site URL _must_ use https or it will throw an error, as PayPal only allows webhooks to go to endpoints using SSL.
